pkg/podman/podman.go

```diff
@@ -0,0 +1,188 @@
+package podman
+
+import (
+	"archive/tar"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/buildah/define"
+	"github.com/containers/podman/v4/pkg/bindings"
+	"github.com/containers/podman/v4/pkg/bindings/containers"
+	"github.com/containers/podman/v4/pkg/bindings/images"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/containers/podman/v4/pkg/specgen"
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"go.uber.org/zap"
+)
+
+const (
+	podmanSock         = "unix:///run/podman/podman.sock"
+	dockerfile         = "Dockerfile"
+	podmanDirName      = "podman"
+	podmanBuildLogFile = "podman-image-build.log"
+)
+
+type Podman struct {
+	context context.Context
+	socket  string
+	out     string
+}
+
+// New setups a podman listening service and returns a connected podman client.
+//
+// Parameters:
+//   - out - location for podman to output any logs created as a result of podman commands
+func New(out string) (*Podman, error) {
+	podmanDirPath := filepath.Join(out, podmanDirName)
+	if err := os.MkdirAll(podmanDirPath, os.ModePerm); err != nil {
+		return nil, fmt.Errorf("creating %s dir: %w", podmanDirPath, err)
+	}
+
+	if err := setupAPIListener(podmanDirPath); err != nil {
+		return nil, fmt.Errorf("creating new podman instance: %w", err)
+	}
+
+	conn, err := bindings.NewConnection(context.Background(), podmanSock)
+	if err != nil {
+		return nil, fmt.Errorf("creating new podman connection: %w", err)
+	}
+
+	return &Podman{
+		context: conn,
+		socket:  podmanSock,
+		out:     podmanDirPath,
+	}, nil
+}
+
+// Import imports a tarball and saves it as a filesystem image
+//
+// Parameters:
+//   - tarball - path to the tarball to be imported
+//   - ref 	  - name for the image that will be created from the tarball
+func (p *Podman) Import(tarball, ref string) error {
+	zap.L().Sugar().Infof("Importing image '%s' from tarball...", ref)
```

Comment:
Nit: If you want to use the suggared logger, you can just call `S()` instead. Also applies to the other occurrences below.
```suggestion
	zap.S().Infof("Importing image '%s' from tarball...", ref)
```